### ISSUE TITLE
[BUG] When obfuscating a py script with a "while" loop, "ERROR invalid v8 code" is reported

### ISSUE THREAD

**Issue by fO-000** [dashingsoft/pyarmor#1011]
Environment:

* Python 3.11.2
* pyarmor 8.0.16 (basic)
* pyarmor.cli.core 1.0.6
* OS: `Linux x 6.1.0-kali5-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.12-1kali2 (2023-02-23) x86_64 GNU/Linux`

The hierarchy of `test_package` is as follows:

```txt
test_package
├── __init__.py
└── __main__.py
```

The content of `__init__.py` is:

```py
#!/usr/bin/env python

```

The content of `__main__.py` is:

```py
#!/usr/bin/env python

while True:
    pass
```

When obfuscating the `test_package`, I got "ERROR invalid v8 code":

```txt
$ pyarmor gen --recursive -i test_package/
INFO     Python 3.11.2
INFO     Pyarmor 8.0.16 (basic), 00xxxx, TBD
INFO     search inputs ...
INFO     find package at test_package
INFO     find 1 top resources
INFO     start to generate runtime files
INFO     target platforms {'linux.x86_64'}
INFO     write dist/test_package/pyarmor_runtime_002329/pyarmor_runtime.so
INFO     patch runtime file
INFO     generate runtime files OK
INFO     start to obfuscate scripts
INFO     process resource "test_package"
INFO     obfuscating file __main__.py
CRITICAL unknown error, please check pyarmor.error.log
ERROR    invalid v8 code
```

The `pyarmor.error.log` is as follows:

```txt
10357 MainProcess 2023-03-26 17:38:38,578
Traceback (most recent call last):
  File "/home/x/.local/lib/python3.11/site-packages/pyarmor/cli/__main__.py", line 567, in main
    main_entry(sys.argv[1:])
  File "/home/x/.local/lib/python3.11/site-packages/pyarmor/cli/__main__.py", line 555, in main_entry
    args.func(ctx, args)
  File "/home/x/.local/lib/python3.11/site-packages/pyarmor/cli/__main__.py", line 147, in cmd_gen
    builder.process(options, pack=args.pack)
  File "/home/x/.local/lib/python3.11/site-packages/pyarmor/cli/generate.py", line 153, in process
    self._obfuscate_scripts()
  File "/home/x/.local/lib/python3.11/site-packages/pyarmor/cli/generate.py", line 118, in _obfuscate_scripts
    code = Pytransform3.generate_obfuscated_script(self.ctx, r)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/x/.local/lib/python3.11/site-packages/pyarmor/cli/core/__init__.py", line 48, in generate_obfuscated_script
    return m.generate_obfuscated_script(ctx, res)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<maker>", line 713, in generate_obfuscated_script
  File "<maker>", line 517, in process
  File "<maker>", line 517, in <listcomp>
  File "<maker>", line 1790, in handle
  File "<maker>", line 1782, in visit
  File "<maker>", line 1750, in patch_co_object
  File "<maker>", line 1472, in _patch_co_object
  File "<maker>", line 1334, in _patch_co_code_py11
RuntimeError: invalid v8 code
```

**Comment by jondy** [comment]
Got it, thanks, your report is terrific, I love it.

**Comment by jondy** [comment]
Fixed in 8.0.17

**Comment by fO-000** [comment]
The issue is confirmed to be resolved. You are also very efficient at solving problems. 👍
